CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift

```diff
@@ -97,9 +98,9 @@ public class PeerToPeerRequestHandler {
                 print("========== Setting CreateIdentity =========")
             }
 
-             if let auth: ListenerAuthenticator = args.get(name: "basic_auth") {
-                config.authenticator = auth
-             }
+            if auth != nil {
+               config.authenticator = auth
+            }
```

Comment:
`args.get(name: "basic_auth")!` this is forced unwrapping variable above, returned value shouldn't be a `nil` right? 

what was wrong with the previous `if let` condition?

Reply:
the previous if let didn't set the authenticator, I am lack of experience with swift, not sure why it caused authenticator a null value (shown in log)

Reply:
basic_auth may or may not be set, if not basic auth method is used (instead, may be tls or something else), then config.authenticator will not be set

Reply:
if the `basic_auth` is not set, is it going to be `(null)` or `nil`? both have different meaning.  From the above I assume, its '(null)', then it's okay to force unwrap. 

Still, I am not sure, how (i)assigning to a variable and then check for nil, vs (ii)if let declaration,  is different. For now, we can keep this way, but I would recommend we could look at this together may be after Lithium?

Reply:
let me schedule a quick chat in zoom, I have another question to ask you.